robot/basestation/static/js/elements/feed/camera.js

```diff
@@ -14,204 +14,269 @@ $(document).ready(() => {
+    const stop_recording_url = '/stop_feed_recording?stream_url=' + stream_url
+    $.ajax(stop_recording_url, {
+      success: data => {
+        appendToConsole(data.msg);
+        callback(data);
+      }, 
+      error: (jqXHR, exception) => {
+        flaskError(jqXHR, exception, stop_recording_url);
+      }
+    });
+  }
+
+
+  function setStreamSelection (availableStreams) {
+    $('.camera-selections').children().empty();
+    availableStreams.forEach(elt => {
+      $('.camera-selections').append(
+        '<li class="camera-selection-element">' + elt + '</li>'
       )
+    })
+  }
+
+  function getStreamURL(topicName) {
+    return 'http://' + getRoverIP() + ':8080/stream?topic=/' + topicName + '/image_raw'
+  }
+
+  function getCameraFilename(cameraPanel) {
+    let cameraName = getCameraName(cameraPanel);
+    let cameraNameSplit = cameraName.split('/');
+    let cameraFilename = cameraNameSplit[cameraNameSplit.length - 1]
+    return cameraFilename;
+  }
+
+  function getCameraName(cameraPanel){
+    let cameraNameElement = cameraPanel.find(".camera-name");
+    let cameraName = cameraNameElement.attr("stream");
+    return cameraName;
+  }
+
+  function showStreamOn(cameraPanel) {
+    let cameraFeed = cameraPanel.find(".camera-feed");
+    let cameraPower = cameraPanel.find('.camera-power');
+
+    cameraFeed.attr("src", getStreamURL(getCameraFilename(cameraPanel) + 'Cam'));
+    cameraFeed.css("padding", "0px");
+
+    cameraPower.attr("power-on", "true");
+    cameraPower.attr("src", POWER_ON);
+  }
+
+  function showStreamOff(cameraPanel) {
+    let cameraFeed = cameraPanel.find('.camera-feed');
+    let cameraPower = cameraPanel.find('.camera-power');
+
+    cameraFeed.attr("src", STREAM_OFF);
+    cameraFeed.css("padding", "10px");
+  
+    cameraPower.attr("power-on", "false");
+    cameraPower.attr("src", POWER_OFF);
+  }
+
+  $('.camera-screenshot').click(e => {
+    let cameraPanel = $(e.target).parents('.camera-panel');
+    let cameraNameElement = cameraPanel.find(".camera-name");
+    let cameraName = cameraNameElement.attr("stream");
+    let cameraStreamName = getCameraFilename(cameraPanel) + 'Cam'
+    let cameraPower = cameraPanel.find('.camera-power');
+
+    if (getCameraName(cameraPanel) == "" || cameraPower.attr("power-on") == "false"){
+      appendToConsole("Please turn on a stream");
+      return;
     }
-  })
 
-  $('#save-image').on('click', function (event) {
-    $.ajax('/capture_image', {
+    $.ajax('/capture_image?stream_url=' + getStreamURL(cameraStreamName), {
       success: function (data) {
         appendToConsole(data.msg)
-        if (!data.msg.includes('success')) {
-          appendToConsole('Something went wrong, got', data.msg)
-        } else {
-          appendToConsole(data.msg)
-        }
-      },
+        },
       error: function () {
-        console.log('An error occured')
+        appendToConsole('An error occured while taking a screenshot')
```

Comment:
Shouldn't we use flaskError to handle this error? It would give more insight on what went wrong.

Reply:
Well the appendToConsole() function sends it to the operators (which in our current state it is software members). Perhaps one day it won't be software members but trained people or members of other divisions. It's much clearer for them "Error while taking a screenshot" then a huge copypasta of programming jargon.

Reply:
I see makes sense.